lib/go-tc/servers.go

```diff
@@ -71,6 +80,12 @@ type ServerDetailV30 struct {
 	ServerInterfaces *[]ServerInterfaceInfo `json:"interfaces"`
 }
 
+// ServerDetailV40 is the details for a server for API v4
+type ServerDetailV40 struct {
```

Comment:
Why not just have this one type declaration instead of a "base" and this combination?